flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SchedulerBase.java

```diff
@@ -641,6 +647,32 @@ public void cancel() {
         return executionGraph.getTerminationFuture().thenApply(FunctionUtils.nullFn());
     }
 
+    protected void archiveGlobalFailure(Throwable failure) {
+        taskFailureHistory.add(new ErrorInfo(failure, System.currentTimeMillis()));
```

Comment:
I would suggest to take the `ExecutionGraph.getStatusTimestamp`. That way the timestamp will be consistent.